src/test/java/org/opensearch/flowframework/FlowFrameworkPluginTests.java

```diff
@@ -8,8 +8,40 @@
  */
 package org.opensearch.flowframework;
 
+import org.opensearch.client.AdminClient;
+import org.opensearch.client.Client;
 import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class FlowFrameworkPluginTests extends OpenSearchTestCase {
-    // Add unit tests for your plugin
+
+    private Client client;
+    private ThreadPool threadPool;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        client = mock(Client.class);
+        when(client.admin()).thenReturn(mock(AdminClient.class));
+        threadPool = new TestThreadPool(FlowFrameworkPluginTests.class.getName());
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        ThreadPool.terminate(threadPool, 500, TimeUnit.MILLISECONDS);
+        super.tearDown();
+    }
+
+    public void testPlugin() throws IOException {
+        try (FlowFrameworkPlugin ffp = new FlowFrameworkPlugin()) {
+            assertEquals(2, ffp.createComponents(client, null, threadPool, null, null, null, null, null, null, null, null).size());
```

Comment:
Why the size is 2 here?

Reply:
Ah! It's from the actual `createComponent`. You can test it something like:
```java
Collection<Object> components = plugin.createComponents(params here)
assertEquals(1, components.size());
assertEquals(threadPool, OpenSearchResources.INSTANCE.getThreadPool());

Reply:
> Ah! It's from the actual `createComponent`. You can test it something like:
> 
> ```java
> Collection<Object> components = plugin.createComponents(params here)
> assertEquals(2, components.size());
> ```

Isn't that exactly what I'm doing?